.githooks/check-format

```diff
@@ -5,6 +5,32 @@
 # This script checks the format of the code and cmake files.
 # In many cases it will automatically fix the issues and abort the commit.
 
+no_formatted_files_staged() {
+    staged_files=$(git diff-index --cached --name-only HEAD)
+    for file in $staged_files; do
+        extension=${file##*.}
+        if [[ "$extension" =~ (h|hpp|c|cpp)$ ]]; then
```

Comment:
Maybe instead check directories of changed files? I think we need to check format only for `src`, `tests` and `cmake`.

Reply:
Cleans it up and _should_ suit the situation.